netket/utils/numbers.py

```diff
@@ -0,0 +1,38 @@
+from numbers import Number
+
+from typing import Union, Any
+from plum import dispatch
+
+import numpy as np
+import jax
+import jaxlib
+
+ArrayT = {np.ndarray, jaxlib.xla_extension.DeviceArray, jax.core.Tracer}
```

Comment:
I think `plum` supports `Union`, so why not use it and make this a proper "pythonic" type?
```suggestion
ArrayT = Union[np.ndarray, jaxlib.xla_extension.DeviceArray, jax.core.Tracer]
```

Reply:
Actually not yet.
That's plum master, which will lead to a new release soon ™.
The released version does not support typing.Union

Reply:
Ah, I see. Maybe we can make a mental note (or, actually better, a comment) to change this once `plum` has a new release.

Reply:
I already did :)

Reply:
Indeed, just saw it.